pymc3/variational/advi.py

```diff
@@ -25,39 +20,46 @@
 def check_discrete_rvs(vars):
     """Check that vars not include discrete variables, excepting ObservedRVs.
     """
-    vars_ = [var for var in vars if not isinstance(var, ObservedRV)]
-    if any([var.dtype in discrete_types for var in vars_]):
+    vars_ = [var for var in vars if not isinstance(var, pm.model.ObservedRV)]
+    if any([var.dtype in pm.discrete_types for var in vars_]):
         raise ValueError('Model should not include discrete RVs for ADVI.')
 
 
+def gen_random_state():
+    """Helper to generate a random state for MRG_RandomStreams"""
+    M1 = 2147483647
+    M2 = 2147462579
+    return np.random.randint(0, M1, 3).tolist() + np.random.randint(0, M2, 3).tolist()
+
+
 def advi(vars=None, start=None, model=None, n=5000, accurate_elbo=False,
-         optimizer=None, learning_rate=.001, epsilon=.1, random_seed=20090425,
+         optimizer=None, learning_rate=.001, epsilon=.1, random_seed=None,
```

Comment:
mentioned above -- `-1` is a valid random seed for this generator, but not for numpy.random.seed.  The opposite is true for `None`.  I believe there's an issue in theano to bring their random number generator API in line with numpy's.